integration_tests/gdk/common/test_recipe_file/test_CaseInsensitiveRecipeFile.py

```diff
@@ -0,0 +1,63 @@
+from pathlib import Path
+from unittest import TestCase
+import tempfile
+import pytest
+
+from gdk.common.recipe_file.CaseInsensitiveRecipeFile import CaseInsensitiveRecipeFile
+
+from requests.structures import CaseInsensitiveDict
+
+
+class CaseInsensitiveRecipeFileTest(TestCase):
```

Comment:
validate that when we read in the recipe and write it out, it is unchanged (we don't want all lowercase keys for example)

Reply:
Added a test that compares the file contents of original yaml and updated yaml.